utils/rpms/daos.spec

```diff
@@ -216,10 +219,14 @@ This is the package needed to build software with the DAOS library.
 %{scons_exe} %{?_smp_mflags} \
       --config=force         \
       --no-rpath             \
-      USE_INSTALLED=all      \
```

Comment:
You cannot remove this from the default list of `scons` arguments as it's required to build daos with external components.

A spec file must build usable RPMs in their "typical configuration" without any external definitions required.  You can only add/change optional behaviour with external options/defines, not require external definitions to get default behaviour.

Reply:
USE_INSTALLED=all should still be set in the regular rpm build flow without coverage.
%{?compiler_args} will set this if coverage if off. But when turned on, we are going to need to build all src with cov, and not use pre-installs.

Reply:
hmmm... reading your comment again, did you mean to say that USE_INSTALLED has to be hardcoded here regardless and cannot be determined via macro?

Reply:
> USE_INSTALLED=all should still be set in the regular rpm build flow without coverage.

How?

> %{?compiler_args} will set this if coverage is off.

I don't see that anywhere in the `daos.spec` file.  It is not allowed to require that this be set externally (i.e. by `--define`) for the default build.

> But when turned on, we are going to need to build all src with cov, and not use pre-installs.

Are you saying that bullseye requires that the entire daos-stack be built with bullseye?  If so, why?  And if so, then why don't other libraries that dynamically link to daos (like glibc for a very obvious example, but will apply to any other dynamic library supplied by the O/S for example) need to be built with bullseye?  Surely that cannot be a requirement.

So assuming it is not a requirement for glibc and all of the other O/S dynamic libraries, it should not be a requirement for our component/dependency libraries.

Additionally, adding component/dependency libraries to the coverage map is going to result in an artificially low coverage value as we don't have tests to test all of the code paths in all of those component/dependency libraries.  You really only want the daos sources to be built with coverage and reporting their testing coverage.

And then there is just the impracticality of trying to include every dependency/component in our RPMs.  It's really just not reasonably feasible.

> hmmm... reading your comment again, did you mean to say that USE_INSTALLED has to be hardcoded here regardless and cannot be determined via macro?

No, it **could** be in a macro, but that macro's default value has to include it.  But per above, the general concept of not using it in the RPM build is just not reasonably feasible, nor will it likely yield desirable results.  So for all practical purposes, it really ought to be hard-coded here.  There is no use-case for not always having it when building RPMs.

Reply:
If USE_INSTALLED is only on daos dependencies, then I think we can leave it here. Will undo this change.